tests/test_aamp_mmotifs.py

```diff
@@ -0,0 +1,248 @@
+    )
+    motif_indices_ref = np.array([[19, 77, 63, 52, 71]])
+    motif_subspaces_ref = [np.array([2])]
+    motif_mdls_ref = [
+        np.array([411.60964047, 423.69925001, 449.11032383, 476.95855027, 506.62406252])
+    ]
+
+    np.random.seed(0)
+    T = np.random.rand(500).reshape(5, 100)
+
+    m = 5
+    excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+    P, I = naive.maamp(T, m, excl_zone)
+    (
+        motif_distances_cmp,
+        motif_indices_cmp,
+        motif_subspaces_cmp,
+        motif_mdls_cmp,
+    ) = aamp_mmotifs(T, P, I)
+
+    npt.assert_array_almost_equal(motif_distances_ref, motif_distances_cmp)
+    npt.assert_array_almost_equal(motif_indices_ref, motif_indices_cmp)
+    npt.assert_array_almost_equal(motif_subspaces_ref, motif_subspaces_cmp)
+    npt.assert_array_almost_equal(motif_mdls_ref, motif_mdls_cmp)
+
+
+@pytest.mark.parametrize("T", test_data)
+def test_aamp_mmotifs_max_distance(T):
+
+    motif_distances_ref = np.array(
+        [[0.0, 1.41421356, 4.46430286, 6.85346628, 8.207923, 8.50529247]]
+    )
+    motif_indices_ref = np.array([[2, 9, 0, 11, 7, 5]])
+    motif_subspaces_ref = [np.array([3])]
+    motif_mdls_ref = [np.array([244.0, 260.67970001, 279.86313714, 281.35940001])]
+
+    m = 4
+    excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+    P, I = naive.maamp(T, m, excl_zone)
+    (
+        motif_distances_cmp,
+        motif_indices_cmp,
+        motif_subspaces_cmp,
+        motif_mdls_cmp,
+    ) = aamp_mmotifs(T, P, I, max_distance=np.inf)
+
+    npt.assert_array_almost_equal(motif_distances_ref, motif_distances_cmp)
+    npt.assert_array_almost_equal(motif_indices_ref, motif_indices_cmp)
+    npt.assert_array_almost_equal(motif_subspaces_ref, motif_subspaces_cmp)
+    npt.assert_array_almost_equal(motif_mdls_ref, motif_mdls_cmp)
+
+
+@pytest.mark.parametrize("T", test_data)
+def test_aamp_motifs_max_distance_max_matches_none(T):
+
+    motif_distances_ref = np.array(
+        [[0.0, 1.41421356, 4.46430286, 6.85346628, 8.207923, 8.50529247]]
+    )
+    motif_indices_ref = np.array([[2, 9, 0, 11, 7, 5]])
+    motif_subspaces_ref = [np.array([3])]
+    motif_mdls_ref = [np.array([244.0, 260.67970001, 279.86313714, 281.35940001])]
+
+    m = 4
+    excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+    P, I = naive.maamp(T, m, excl_zone)
+    (
+        motif_distances_cmp,
+        motif_indices_cmp,
+        motif_subspaces_cmp,
+        motif_mdls_cmp,
+    ) = aamp_mmotifs(T, P, I, max_distance=np.inf, max_matches=None)
+
+    npt.assert_array_almost_equal(motif_distances_ref, motif_distances_cmp)
+    npt.assert_array_almost_equal(motif_indices_ref, motif_indices_cmp)
+    npt.assert_array_almost_equal(motif_subspaces_ref, motif_subspaces_cmp)
+    npt.assert_array_almost_equal(motif_mdls_ref, motif_mdls_cmp)
+
+
+@pytest.mark.parametrize("T", test_data)
+def test_aamp_mmotifs_exactly_one_motif_pair_k_chosen(T):
```

Comment:
Maybe a small suggestion that we could even shorten and make the function name:

```
def test_aamp_mmotifs_max_motifs_1_max_matches_2_k_1(T):
```
This way, just from reading the function name, I can anticipate what parameters matter in this test? What do you think? I'm good either way.